### PR TITLE
Extend documentation of data method for multi-threading

### PR DESCRIPTION
**Pull request by jens-mueller-sociomantic**
The added documentation states that when using more than one MXNet
thread, i.e., when using a threaded engine, calls to `data` must ensure
that pending writes have been finished (when reading) or that pending
reads/writes have been finished (when writing). It is important to
follow the documented advice for multi-threaded engines. Otherwise
garbage may be read which may yield incorrect results.

### CODE REVIEW

File: src/mxnet/NDArray.d [hunk 1]
```diff
@@ -344,6 +344,15 @@ public class NDArray (T)
         `[1, 2, 3,
           4, 5, 6]`.
 
+        Note that it is only safe to read/write this data once any pending
+        read/writes are completed. In a multi-threaded setups (i.e. when using
```
Comment (joseph-wakeling-sociomantic) [inline src/mxnet/NDArray.d]: `In multi-threaded setups` or `In a multi-threaded setup` (your call).

**Review by joseph-wakeling-sociomantic — CHANGES_REQUESTED** [review]
Nice addition, thanks!  One small typo to fix, otherwise LGTM.

**Comment by jens-mueller-sociomantic** [comment]
```diff
diff --git a/src/mxnet/NDArray.d b/src/mxnet/NDArray.d
index 26dc64a..9a41b6b 100644
--- a/src/mxnet/NDArray.d
+++ b/src/mxnet/NDArray.d
@@ -345,7 +345,7 @@ public class NDArray (T)
           4, 5, 6]`.
 
         Note that it is only safe to read/write this data once any pending
-        read/writes are completed. In a multi-threaded setups (i.e. when using
+        read/writes are completed. In a multi-threaded setup (i.e. when using
         a threaded MXNet engine) calling `waitToRead` ensures all writes to
         this slice have been finished. Similarly, when writing elements of the
         slice calling `waitToWrite` ensures that all previous read/writes of
```

**Review by joseph-wakeling-sociomantic — APPROVED** [review]
LGTM.  I'll take in when CI completes.

**Pull request merged**